homeassistant/components/camera/demo.py

```diff
@@ -29,15 +30,21 @@ def __init__(self, hass, config, name):
         self._parent = hass
         self._name = name
         self._motion_status = False
+        self.is_streaming = True
+        self._images = {}
 
     def camera_image(self):
         """Return a faked still image response."""
-        now = dt_util.utcnow()
+        index = 'off'
+        if self.is_streaming:
+            index = str(dt_util.utcnow().second % 4)
 
-        image_path = os.path.join(
-            os.path.dirname(__file__), 'demo_{}.jpg'.format(now.second % 4))
-        with open(image_path, 'rb') as file:
-            return file.read()
+        if index not in self._images:
+            image_path = os.path.join(
+                os.path.dirname(__file__), 'demo_{}.jpg'.format(index))
+            with open(image_path, 'rb') as file:
+                self._images[index] = file.read()
```

Comment:
We should not cache all the images in memory

Reply:
That is for Demo camera only.

Reply:
I know, and we should not do it. We only load an image every 5 seconds and it should not hold memory when not in use.